From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mortyzhaoyu <morty.zhaoyu@gmail.com>
Date: Tue, 24 Jun 2026 16:00:00 +0800
Subject: [PATCH] rk3576-ipa-6.18-compat

Linux 6.18 no longer ships trace/events/thermal.h or thermal_ipa_power.h,
and struct thermal_zone_device is opaque in linux/thermal.h. Remove obsolete
trace code and use thermal_zone_get_temp() / thermal_zone_device_type().

---
 drivers/soc/rockchip/rockchip_ipa.c | 16 +++-------------
 1 file changed, 3 insertions(+), 13 deletions(-)

--- a/drivers/soc/rockchip/rockchip_ipa.c
+++ b/drivers/soc/rockchip/rockchip_ipa.c
@@ -9,10 +9,6 @@
 #include <linux/thermal.h>
 #include <soc/rockchip/rockchip_ipa.h>
 #include <soc/rockchip/rockchip_opp_select.h>
-#include <trace/events/thermal.h>
-
-#define CREATE_TRACE_POINTS
-#include <trace/events/thermal_ipa_power.h>
 
 #define FALLBACK_STATIC_TEMPERATURE 55000
 
@@ -203,10 +199,10 @@
 	if (!data)
 		return 0;
 
-	ret = data->tz->ops->get_temp(data->tz, &temp);
+	ret = thermal_zone_get_temp(data->tz, &temp);
 	if (ret) {
 		pr_err("%s:failed to read %s temp\n",
-		       __func__, data->tz->type);
+		       __func__, thermal_zone_device_type(data->tz));
 		temp = FALLBACK_STATIC_TEMPERATURE;
 	}
 
@@ -218,14 +214,6 @@
 	power_big = (u64)static_power * (u64)volt_scaling_factor;
 	static_power = div_u64(power_big, 1000000);
 
-	trace_thermal_ipa_get_static_power(data->leakage,
-					   data->static_coefficient,
-					   temp,
-					   temp_scaling_factor,
-					   (u32)voltage_mv,
-					   volt_scaling_factor,
-					   static_power);
-
 	return static_power;
 }
 EXPORT_SYMBOL(rockchip_ipa_get_static_power);
